packages/sales-api-service/src/services/exceptions/__tests__/pocl-validation-errors.service.spec.js

```diff
@@ -0,0 +1,170 @@
+  amount: payload.finaliseTransactionPayload.payment.amount,
+  channelId: payload.finaliseTransactionPayload.payment.channelId,
+  paymentSource: payload.finaliseTransactionPayload.payment.source,
+  methodOfPayment: payload.finaliseTransactionPayload.payment.method,
+  status: 'Ready for Processing',
+  errorMessage: payload.errorMessage
+})
+
+describe('POCL validation error service', () => {
+  beforeEach(jest.clearAllMocks)
+
+  describe('createPoclValidationError', () => {
+    let payload
+    beforeEach(async () => {
+      payload = getPayload()
+      await createPoclValidationError(payload)
+    })
+
+    it('maps the record to an instance of PoclDataValidationError', async () => {
+      const [poclValidationError] = persist.mock.calls[0][0]
+      expect(poclValidationError).toBeInstanceOf(PoclValidationError)
+    })
+
+    it('creates the validation record', async () => {
+      const [poclValidationError] = persist.mock.calls[0][0]
+      expect(poclValidationError).toMatchSnapshot()
+    })
+  })
+
+  describe('updatePoclValidationError', () => {
+    let payload
+    beforeEach(async () => {
+      payload = getPayload()
+      delete payload.createTransactionError
+      payload.errorMessage = 'Invalid email address'
+    })
+
+    describe('when validation error record exists', () => {
+      let poclValidationError
+      beforeEach(async () => {
+        findById.mockResolvedValue(getValidationError(payload))
+        await updatePoclValidationError('pocl-validation-error-id', payload)
+        poclValidationError = persist.mock.calls[0][0][0]
+      })
+
+      it('retrieves existing record', async () => {
+        expect(findById).toBeCalledWith(PoclValidationError, 'pocl-validation-error-id')
+      })
+
+      describe('and status is not provided', () => {
+        it('sets the status to "Needs Review"', async () => {
+          expect(poclValidationError.status.label).toBe('Needs Review')
+        })
+
+        it('the state code is not set', async () => {
+          expect(poclValidationError.stateCode).toBe(undefined)
+        })
+
+        it('updates the validation record', async () => {
+          expect(poclValidationError).toMatchSnapshot()
+        })
+      })
+
+      describe('and status equals "Processed"', () => {
+        beforeEach(async () => {
+          payload.status = 'Processed'
+          await updatePoclValidationError('pocl-validation-error-id', payload)
+        })
+
+        it('the status is set to "Processed"', async () => {
+          const [poclValidationError] = persist.mock.calls[0][0]
+          expect(poclValidationError.status.label).toBe('Processed')
+        })
+
+        it('the state code is set to 1', async () => {
+          const [poclValidationError] = persist.mock.calls[0][0]
+          expect(poclValidationError.stateCode).toBe(1)
+        })
+
+        it('updates the validation record', async () => {
+          const [poclValidationError] = persist.mock.calls[0][0]
```

Comment:
yeah, here too (as well as the two above), plus any others in the file. I'd say go for the first option (`const [[[poclValidationError]]]`) as it's a little more succinct.